src/arch/arm/Mir.zig

```diff
@@ -0,0 +1,218 @@
+        /// Push multiple registers to Stack
+        push,
+        /// Reverse Subtract
+        rsb,
+        /// Store Register
+        str,
+        /// Store Register Byte
+        strb,
+        /// Store Register Halfword
+        strh,
+        /// Subtract
+        sub,
+        /// Supervisor Call
+        svc,
+    };
+
+    /// The position of an MIR instruction within the `Mir` instructions array.
+    pub const Index = u32;
+
+    /// All instructions have a 4-byte payload, which is contained within
+    /// this union. `Tag` determines which union field is active, as well as
+    /// how to interpret the data within.
+    pub const Data = union {
+        /// No additional data
+        ///
+        /// Used by e.g. nop
+        nop: void,
+        /// Another instruction
+        ///
+        /// Used by e.g. b
+        inst: Index,
+        /// A 16-bit immediate value.
+        ///
+        /// Used by e.g. bkpt
+        imm16: u16,
+        /// A 24-bit immediate value.
+        ///
+        /// Used by e.g. svc
+        imm24: u24,
+        /// Index into `extra`. Meaning of what can be found there is context-dependent.
+        ///
+        /// Used by e.g. load_memory
+        payload: u32,
+        /// A register
+        ///
+        /// Used by e.g. blx
+        reg: Register,
+        /// A register and a 16-bit unsigned immediate
+        ///
+        /// Used by e.g. movw
+        r_imm16: struct {
+            rd: Register,
+            imm16: u16,
+        },
+        /// Two registers and a shift amount
+        ///
+        /// Used by e.g. lsl
+        rr_shift: struct {
+            rd: Register,
+            rm: Register,
+            shift_amount: bits.Instruction.ShiftAmount,
+        },
+        /// Two registers and an operand
+        ///
+        /// Used by e.g. sub
+        rr_op: struct {
+            rd: Register,
+            rn: Register,
+            op: bits.Instruction.Operand,
+        },
+        /// Two registers and an offset
+        ///
+        /// Used by e.g. ldr
+        rr_offset: struct {
+            rt: Register,
+            rn: Register,
+            offset: bits.Instruction.OffsetArgs,
+        },
+        /// Two registers and an extra load/store offset
+        ///
+        /// Used by e.g. ldrh
+        rr_extra_offset: struct {
+            rt: Register,
+            rn: Register,
+            offset: bits.Instruction.ExtraLoadStoreOffsetArgs,
+        },
+        /// Three registers
+        ///
+        /// Used by e.g. mul
+        rrr: struct {
+            rd: Register,
+            rn: Register,
+            rm: Register,
+        },
+        /// An unordered list of registers
+        ///
+        /// Used by e.g. push
+        register_list: bits.Instruction.RegisterList,
+        /// Debug info: line and column
+        ///
+        /// Used by e.g. dbg_line
+        dbg_line_column: struct {
+            line: u32,
+            column: u32,
+        },
```

Comment:
This makes `Data` 8 bytes instead of 4 as the doc comments above Data suggest.

Reply:
why not use the commented out comptime block below to assert that the size of Data is what it is expected to be?

Reply:
Yes, `Data` is supposed to be 8 bytes. However, due to the use of tagged unions such as `bits.Instruction.Operand` etc. this is not the case yet and `Data` is currently 12 bytes large. Should I flatten down `Data` to 8 bytes here or can I do that in a follow-up PR?

Reply:
Fine by me to do in a follow-up PR, as long as you see it as a to-do item :)

Reply:
Ok, I corrected the doc comment and added a TODO comment for fixing the size of `Data` in future.